pybamm/solvers/rk4_ode_solver.py

```diff
@@ -0,0 +1,186 @@
+#
+# Solver class using fixed step Runge-Kutta 4 integrator
+#
+import casadi
+import pybamm
+import numpy as np
+
+
+class RK4Solver(pybamm.BaseSolver):
+    """Solve a discretised model, using RK4.
+
+    **Extends**: :class:`pybamm.BaseSolver`
+
+    Parameters
+    ----------
+    N : float
+        The number integration steps within a time step.
+
+    """
+
+    def __init__(
+        self,
+        N,  # number of control intervals
+        rtol=1e-6,
+        atol=1e-6,
+        root_method="casadi",
+        root_tol=1e-6,
+        max_step_decrease_count=20,
+    ):
+        super().__init__("problem dependent", rtol, atol, root_method, root_tol)
+        self.N = N
+        self.name = "RK4 ODE solver"
+
+        self.max_step_decrease_count = max_step_decrease_count
+
+    def _integrate(self, model, t_eval, inputs=None):
+        """
+        Solve a model defined by dydt with initial conditions y0.
+
+        Parameters
+        ----------
+        model : :class:`pybamm.BaseModel`
+            The model whose solution to calculate.
+        t_eval : numeric type
+            The times at which to compute the solution
+        inputs : dict, optional
+            Any input parameters to pass to the model when solving
+
+        """
+        inputs = inputs or {}
+        # convert inputs to casadi format
+        inputs = casadi.vertcat(*[x for x in inputs.values()])
+
+        integrator, F = self.get_integrator(model, t_eval, inputs)
+        solution = self._run_integrator(integrator, model, model.y0, inputs, t_eval)
+        solution.F = F
+        return solution
+
+    def get_integrator(self, model, t_eval, inputs):
+        # Only set up problem once
+        y0 = model.y0
+        if isinstance(y0, casadi.DM):
+            y0 = y0.full().flatten()
+        rhs = model.casadi_rhs
+        # algebraic = model.casadi_algebraic
```

Comment:
what if this solver is used on a dae model? Should fail gracefully with some sort of error message

Reply:
@pmohtat you can set `self.ode_solver=True` in the init function to do this check automatically, and remove this commented-out line